src/QMCDrivers/WFOpt/QMCCostFunctionBatched.cpp

```diff
@@ -360,14 +376,44 @@ void QMCCostFunctionBatched::checkConfigurations()
 
         for (int ib = 0; ib < current_batch_size; ib++)
         {
+            #ifdef HAVE_LMY_ENGINE
```

Comment:
more ifdefs, this code smells and bad. 

It looks like I do this extra allocation and setup even if I'm not using the LMY engine as long as the compile switch is on.  Ifdef's should be temporary stopgap solutions and not a way to shove low quality code into the codebase. I'm also no longer sure where this method started or what exactly we are trying to accomplish.